Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp

```diff
@@ -42,6 +45,17 @@ MappingButton::MappingButton(MappingWidget* widget, ControlReference* ref, bool
     : ElidedButton(EscapeAmpersand(QString::fromStdString(ref->GetExpression()))), m_parent(widget),
       m_reference(ref)
 {
+  // Force all mapping buttons to use stay at a minimal height
+  int height = QFontMetrics(qApp->font()).height() + 2 * VERTICAL_PADDING;
+
+  setMinimumHeight(height);
+  setMaximumHeight(height);
+
+  // Make sure that long entries don't throw our layout out of whack
+  setMaximumWidth(115);
+
+  setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Expanding);
```

Comment:
Do you have an example where the button would need to vertically expand? Then again, you set Min/Max Height to the same value, so this will never happen (unless Qt is kinda special with those values).

It used to be horizontal expand/vertical fixed instead of horizonal fixed/vertical expand before; and I think neither makes really much sense here (preferring fixed size and some ellipsis for longer expressions)